app/models/person.rb

```diff
@@ -1,6 +1,74 @@
 class Person < ActiveRecord::Base
-  has_one :location
+  has_one :location, autosave: true
   has_one :district, through: :location
+  has_one :representative, through: :district
+  has_one :target_rep, -> { targeted }, through: :district
+  has_one :state, through: :location
+  has_many :senators, through: :state
 
-  validates :email, presence: true, uniqueness: { case_sensitive: false }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, uniqueness: { case_sensitive: false },
+                    format: { with: VALID_EMAIL_REGEX }
+
+  before_save { self.email = email.downcase }
+
+  alias_method :location_association, :location
```

Comment:
the location method just passes everything to the location association, so there shouldn't be extra work. The main thing this does is allow me to set someone's district, etc., without first checking if they have an associated location. The other common option would be to have an after_create callback on person to create their associated location, but I liked this better